### PR TITLE
Fix move popup to use jquery.position, fix page jumping

### LINKED ISSUE #621: move popup sometimes causes page to jump
- open move popup on a primary item,
- scroll to emotes 
- open a popup on one of those 

page jumps for a split second.

http://i.imgur.com/v2Sdyz6.gifv

### PR DESCRIPTION
**Pull request by bhollis**
This fixes the "jumping" that happened when clicking items (it was caused by moving the input focus around), and positions the dialog such that it won't ever be offscreen. The major differences now:

1. No more page jumping.
2. There's a drop shadow.
3. The dialog is slightly closer to the item.
4. The dialog prefers showing up below the item, rather than above it (which was the preference before). It's easier to scroll down to see more info.
5. The dialog will never be offscreen or hidden.
6. In the minmax tool, the dialog will always show up next to the thing you clicked, rather than the first instance of it in the page. This is because we no longer use the element ID to position the dialog.

On the dev side, we've got some nice properties such as the popup being attached to `body` instead of the item (which means it won't accidentally inherit styles, etc).

I'll add on a change to this that moves the move popup into its own service (like ngDialog) so it can be safely used from all kinds of places.

This fixes #621, #539, and #508.

### CODE REVIEW

**Comment by bhollis** [comment]
Hmm, I played with this a bit, and I think I do like the previous behavior of having the popup prefer being above the item, because otherwise the buttons are hard to hit. So I put that back.

What do you think of something like this for when the popup is below, though:
screen shot 2016-07-04 at 10 43 52 pm

**Comment by kyleshay** [comment]
Awesome Ben!! :D The style updates are especially nice.  

Per the move options above the title -- a lot less distance for the mouse to travel would be nice. Though, moving them does seem a bit strange. I'm not entirely sure how often people are clicking the buttons versus dragging and dropping anyway.

Another option may be to have the buttons just below the title bar (IFF the dialog opens "down".) That's how the move popup looks if the stats/perks are collapsed.

screen shot 2016-07-05 at 9 23 15 am

One minor issue, if the move popup is expanded or collapsed, it does not reposition the dialog.

demop

demop2

**Comment by kyleshay** [comment]
nitpick/opinion...that may have been better suited for the redesign issue.

when the popup opens "up", the chevron seems to be facing the "wrong" direction. not sure what other language we could use there.

**Comment by bhollis** [comment]
Good points, I'll tweak those. The "buttons on top" was just a sketch, I
like your version with them under the header better.

**Comment by bhollis** [comment]
OK, here's a fix for the changing height, but it'm not 100% happy with it. The code feels pretty hacky. It fixes the problem but I really wish CSS would just let you position things relative to other elements directly...

Two weird things now:
1. There's a little flash when you expand and collapse the details. This is because I have to use a timeout to set the position since the DOM changes haven't applied yet when my function is called.
2. When you collapse a properties dialog that extends downward, it may suddenly have space up top and thus switch to displaying above the item. That makes sense, sorta, but it's weird.

File: app/scripts/xur/dimXur.controller.js [hunk 1]
```diff
@@ -33,17 +33,6 @@
       countCurrencies();
     });
 
-    $scope.$on('ngDialog.opened', function(event, $dialog) {
```
Comment (kyleshay) [inline app/scripts/xur/dimXur.controller.js]: did we not need positioning logic in here anymore?

**Comment by kyleshay** [comment]
> 1. There's a little flash when you expand and collapse the details. This is because I have to use a timeout to set the position since the DOM changes haven't applied yet when my function is called.

Bit funky, but it's better than current behavior... nice trick with the timeout. Collapsing seems fine, the only strange case is if you're clicking on an item with the minimized popup above... and then expand the dialog, it goes from top -> below (since it's still rendering?) -> back to top (since it had space.)

Tried playing with some things but wasn't able to get far.

> 2. When you collapse a properties dialog that extends downward, it may suddenly have space up top and thus switch to displaying above the item. That makes sense, sorta, but it's weird.

Totally fine with that and it looks great!

**Comment by bhollis** [comment]
Yeah this is a case where if I was just doing regular JS I could make it a
lot smoother but having to work within angular makes it super tough.

**Comment by kyleshay** [comment]
Would be a bit of additional work, but if we moved the move options below the header so there is less mouse movement like you were suggesting... 

we could just always show the expanded popup, and remove all settings/options around collapsing it.

**Comment by bhollis** [comment]
Let's merge this but keep talking about improvements.

**Comment by bhollis** [comment]
I'm a little shy about ever removing options right now

**Comment by kyleshay** [comment]
:( ... on the bright side, there hasn't been a single complaint about removing the option for showing the primary stat on items... so that was a great call! 

I only personally use the move popups for moving multiple items or seeing stats... so I always have it expanded. I wonder how many use the move popup for just moving items.... and not seeing the options. :/ worth maybe doing some testing.

**Pull request merged**